libioc/Jail.py

```diff
@@ -1832,39 +1660,69 @@ def get_hook_script_path(self, hook_name: str) -> str:
         """Return the absolute path to the hook script file."""
         return f"{self.jail.launch_script_dir}/{hook_name}.sh"
 
-    def _start_vimage_network(self) -> typing.Tuple[
-        'libioc.Network.CreatedCommandList',
-        'libioc.Network.JailCommandList'
-    ]:
-        self.logger.debug("Starting VNET/VIMAGE")
-
-        created: typing.List[str] = []
-        start: typing.List[str] = []
+    def _start_vimage_network(
+        self,
+        event_scope: typing.Optional['libioc.events.Scope']=None
+    ) -> typing.Generator['libioc.events.VnetInterface', None, None]:
+        
+        event = libioc.events.VnetSetup(
+            jail=self,
+            scope=event_scope
+        )
+        yield event.begin()
 
         for network in self.networks:
-            _created, _start = network.setup()
-
-            created += _created
-            start += _start
+           yield from network.setup(event_scope=event.scope)
 
-        return created, start
+        yield event.end()
 
     def __stop_network(
         self,
         jid: int,
         force: bool,
         event_scope: 'libioc.events.Scope'
-    ) -> typing.Generator['libioc.events.JailHook', None, None]:
+    ) -> typing.Generator['libioc.events.IocEvnet', None, None]:
+        
+        if self.config["vnet"] is True:
+            yield from self._stop_vimage_network(jid, event_scope=event_scope)
+        else:
+            # ToDo: clarify why this step is wanted
+            return
+            yield from self._stop_non_vimage_network(
+                force=force,
+                event_scope=event_scope
+            )
+
+    def _stop_non_vimage_network(
```

Comment:
on the one hand i'm thinking, there has to be a better name for this
but on the other hand, this kind of shows our preference ;)

Reply:
What should it be called? Regular Network? 🤷‍♂️

Reply:
well, it's what other containers and virtualuzations call a host-network, but that, from inside this code, can be confusing